In [1]:
import pandas as pd
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_squared_error, r2_score

In [2]:
# Load the training and testing datasets

train_df = pd.read_csv('C:/Users/xpert/Downloads/churn-bigml-80.csv')
test_df = pd.read_csv('C:/Users/xpert/Downloads/churn-bigml-20.csv')

print(train_df.head())

  State  Account length  Area code International plan Voice mail plan  \
0    KS             128        415                 No             Yes   
1    OH             107        415                 No             Yes   
2    NJ             137        415                 No              No   
3    OH              84        408                Yes              No   
4    OK              75        415                Yes              No   

   Number vmail messages  Total day minutes  Total day calls  \
0                     25              265.1              110   
1                     26              161.6              123   
2                      0              243.4              114   
3                      0              299.4               71   
4                      0              166.7              113   

   Total day charge  Total eve minutes  Total eve calls  Total eve charge  \
0             45.07              197.4               99             16.78   
1             27.47   

In [3]:
print(test_df.head())

  State  Account length  Area code International plan Voice mail plan  \
0    LA             117        408                 No              No   
1    IN              65        415                 No              No   
2    NY             161        415                 No              No   
3    SC             111        415                 No              No   
4    HI              49        510                 No              No   

   Number vmail messages  Total day minutes  Total day calls  \
0                      0              184.5               97   
1                      0              129.1              137   
2                      0              332.9               67   
3                      0              110.4              103   
4                      0              119.3              117   

   Total day charge  Total eve minutes  Total eve calls  Total eve charge  \
0             31.37              351.6               80             29.89   
1             21.95   

In [4]:
# Quick validation

print("Missing values per column:\n", train_df.isnull().sum())
print(train_df.duplicated().sum())

Missing values per column:
 State                     0
Account length            0
Area code                 0
International plan        0
Voice mail plan           0
Number vmail messages     0
Total day minutes         0
Total day calls           0
Total day charge          0
Total eve minutes         0
Total eve calls           0
Total eve charge          0
Total night minutes       0
Total night calls         0
Total night charge        0
Total intl minutes        0
Total intl calls          0
Total intl charge         0
Customer service calls    0
Churn                     0
dtype: int64
0


In [5]:
print("Missing values per column:\n", test_df.isnull().sum())
print(test_df.duplicated().sum())

Missing values per column:
 State                     0
Account length            0
Area code                 0
International plan        0
Voice mail plan           0
Number vmail messages     0
Total day minutes         0
Total day calls           0
Total day charge          0
Total eve minutes         0
Total eve calls           0
Total eve charge          0
Total night minutes       0
Total night calls         0
Total night charge        0
Total intl minutes        0
Total intl calls          0
Total intl charge         0
Customer service calls    0
Churn                     0
dtype: int64
0


In [6]:
# Predict Total day charge from Total day minutes

x_train = train_df[['Total day minutes']]
y_train = train_df['Total day charge']

x_test = test_df[['Total day minutes']]
y_test = test_df['Total day charge']

In [7]:
# Create and Fit the model

model = LinearRegression()
model.fit(x_train, y_train)

# Predict on the test set

y_pred = model.predict(x_test)

In [8]:
# Coefficients

print("Intercept:", model.intercept_)
print("Slope:", model.coef_[0])

Intercept: 0.000618521836301511
Slope: 0.16999950056381516


Interpretation:  
The intercept (0.0006) represents the baseline Total Day Charge when Total Day Minutes is 0.  
The slope (0.17) means that for every 1-minute increase in Total Day Minutes, the Total Day Charge increases by approximately 0.17 units.


In [10]:
# Evaluation metrics

r2 = r2_score(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)

mse, r2

(8.240420038735245e-06, 0.9999999073207673)

Evaluation: 
The model achieved an R-square of 0.9999999, indicating that almost all the variation in Total Day Charge is explained by Total Day Minutes.  
The mean squared error 8.24e-6 is extremely low, showing very accurate predictions.
